### ISSUE TITLE
Warnings about reading and writing when only reading

### ISSUE THREAD

**Issue by sboyd-m** [koalaman/shellcheck#2325]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2094, SC2094
- My shellcheck version (`shellcheck --version` or 'online'): online
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

while read -r a; do
        read -r b < file.txt
done < file.txt

echo "$a" "$b"
```

#### Here's what shellcheck currently says:
```
Line 4	SC2094: Make sure not to read and write the same file in the same pipeline.
Line 5	SC2094: Make sure not to read and write the same file in the same pipeline.
```
#### Here's what I wanted or expected to see:
No warning.

The warning should only show up if there is a mixture of reads/writes. Multiple reads should always be safe. Multiple writes *might* be safe, I'm not sure about how the flushing behaviors are defined (I doubt they are well defined, so this is probably unsafe).

**Comment by koalaman** [comment]
This warning was already suppressed for multiple redirections (e.g. `>>`), and is now similarly suppressed for two inputs like here. Thanks! 

You still have similar false positives of two reads with something like `cat foo - < foo`, but this is harder for ShellCheck to determine. You can also safely do concurrent appends as they are atomic under a certain size except on NFS etc, but really the check is primarily intended to catch the very common `cat foo | sed 's/a/b/' > foo` type pitfalls.
